modules/l4proxy/matcher.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Matthew Holt
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4proxy
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/mholt/caddy-l4/layer4"
+)
+
+// https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
+var (
+	headerV1Prefix = []byte("PROX") // intentional to not include "Y", see match() function for details
+	headerV2Prefix = []byte{0x0D, 0x0A, 0x0D, 0x0A, 0x00, 0x0D, 0x0A, 0x51, 0x55, 0x49, 0x54, 0x0A}
+)
+
+func init() {
+	caddy.RegisterModule(MatchPROXY{})
+}
+
+type MatchPROXY struct{}
+
+// CaddyModule returns the Caddy module information.
+func (MatchPROXY) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "layer4.matchers.ProxyProtocol",
```

Comment:
I think typically we use `snake_case`

```suggestion
		ID:  "layer4.matchers.proxy_protocol",
```

Reply:
Thanks for raising that, I've changed to snake case now.